model_analyzer/config/generate/model_variant_name_manager.py

```diff
@@ -21,95 +22,78 @@ class ModelVariantNameManager:
 
     def __init__(self):
 
-        # Stored as _variant_names[base_model_name][param_combo] = variant_name_string
-        self._variant_names = {}
+        # Dict of {model_config_name: model_config_dict}
+        self._model_config_dicts: Dict[str, Dict] = {}
 
-        # Stored as _model_name_index[base_model_name] = current_count_integer
-        self._model_name_index = {}
+        # Dict of {base_model_name: current_count_integer}
+        self._model_name_index: Dict[str, int] = {}
 
-    def get_model_variant_name(self, base_model_name, param_combo):
+    @classmethod
+    def from_dict(
+            cls,
+            model_variant_name_manager_dict: Dict) -> "ModelVariantNameManager":
+        model_variant_name_manager = ModelVariantNameManager()
+
+        model_variant_name_manager._model_config_dicts = model_variant_name_manager_dict[
+            '_model_config_dicts']
+        model_variant_name_manager._model_name_index = model_variant_name_manager_dict[
+            '_model_name_index']
+
+        return model_variant_name_manager
+
+    def get_model_variant_name(self, model_name: str, model_config_dict: Dict,
+                               param_combo: Dict) -> Tuple[bool, str]:
         """
         Given a base model name and a dict of parameters to be applied
-        to the base model config, return the name of the model variant
+        to the base model config, return the name of the model variant 
+        and if the variant already existed
 
         If the same input values are provided to this function multiple times, 
         the same value will be returned
         """
-        if self._model_variant_exists(base_model_name, param_combo):
-            return self._get_model_variant_name(base_model_name, param_combo)
-        else:
-            return self._make_model_variant_name(base_model_name, param_combo)
+        mcd = self._copy_and_restore_mcd_name(model_name, model_config_dict)
 
-    def _model_variant_exists(self, base_model_name, param_combo):
-        """ 
-        Returns true if a model variant name already exists for this combination 
-        """
-        hashable_key = self._get_hashable_key(param_combo)
+        variant_found, model_variant_name = self._find_existing_variant(mcd)
 
-        return base_model_name in self._variant_names \
-           and hashable_key in self._variant_names[base_model_name]
+        if variant_found:
+            return (True, model_variant_name)
 
-    def _get_model_variant_name(self, base_model_name, param_combo):
-        """
-        Returns the model variant name that already exists for this combination
-        """
-        hashable_key = self._get_hashable_key(param_combo)
-        return self._variant_names[base_model_name][hashable_key]
+        if self._is_default_config(param_combo):
```

Comment:
it is weird that we are still passing in the param combo for only this reason. Seems like we should be able to completely operate in the model_config_dict space instead of param_combo, but I don't think this class has enough information. Not sure of anything better. Just take a min to look to see if any better solution

Reply:
Without param combo I couldn't figure out how to tell what the default config was from the model_config_dict. And, I even if you could I think it would be even more awkward to figure out.